backend/src/meme/services/meme.service.ts

```diff
@@ -0,0 +1,78 @@
+import { Injectable, StreamableFile } from "@nestjs/common";
+import { InjectRepository } from "@nestjs/typeorm";
+import { Repository } from "typeorm";
+import { Meme } from "../model/meme.entity";
+import { v4 as uuidv4 } from "uuid";
+import { MemeDto } from "../model/dto/meme.dto";
+import { SourceType } from "../model/meme.enums";
+import { ConfigService } from "@nestjs/config";
+import { createReadStream } from "fs";
+import { join } from "path";
+const fs = require("fs");
+@Injectable()
+export class MemeService {
+	constructor(
+		@InjectRepository(Meme) private memeRepository: Repository<Meme>,
+		private configService: ConfigService
+	) {}
+	async addMemeByFile(file): Promise<[Promise<any>, Promise<any>]> {
+		const fileName = file.originalname;
+		const fileExtension = file.originalname.split(".").pop();
+		const fileSource = uuidv4() + "." + fileExtension;
+
+		const fileWritePromise = new Promise((resolve, reject) => {
+			fs.writeFile(
+				`${this.configService.get("FILES_DIRECTORY")}${fileSource}`,
+				file.buffer,
+				function (error) {
+					if (error) {
+						reject(error);
+					}
+					resolve(true);
+				}
+			);
+		});
+
+		const memeDto = new MemeDto(
+			fileName,
+			SourceType.LOCAL_FILE,
+			fileSource,
+			fileExtension
+		);
+
+		const addToDbPromise = new Promise(async (resolve, reject) => {
```

Comment:
U don't need create promise, if u create a operation with repositories, this return u a promise, and when u return it from service, it's still promise.